classes/Datasets/Transforms.sc

```diff
@@ -0,0 +1,122 @@
+/*****************************************
+Data Transforms
+(C) 2018 Jonathan Reus
+
+Tools for analyzing and transforming datasets: scaling, normalization, standardization, PCA, etc..
+
+
+******************************************/
+
+// TO NORMALIZE:
+// find min / max of features
+// normalized_value = (val - min) / (max - min)
+
+
+/*
+T for Transform
+Normalization:
+1. find min/max of each feature
+2. normalized_value = (val - min) / (max - min)
+*/
+TNormalizer {
+	var <min, <max, <originalData, <normalizedData, dim;
+
+	/*
+	@param dataset a Matrix of rows as feature vectors
+	*/
+	*new {|dataset|
+		^super.new.init(dataset);
+	}
+
+	init {|dataset|
+		if(dataset.isKindOf(SequenceableCollection).not ) { "Dataset must be a Matrix".error;this.halt; };
```

Comment:
Better write:

```
Error("Dataset must be a Matrix".).throw
```

Reply:
btw. the test `.isKindOf(SequenceableCollection)` doesn't exclude things like a `Set`. So if you pass a `Set` it would go through.